tests/test_dashboard_client.cpp

```diff
@@ -219,6 +219,33 @@ TEST_F(DashboardClientTest, connect_non_running_robot)
   EXPECT_LT(elapsed, 2 * comm::TCPSocket::DEFAULT_RECONNECTION_TIME);
 }
 
+TEST_F(DashboardClientTest, non_expected_result_returns_correctly)
+{
+  std::unique_ptr<DashboardClient> dashboard_client;
+  dashboard_client.reset(new DashboardClient("192.168.56.101"));
+  ASSERT_TRUE(dashboard_client->connect());
+  EXPECT_TRUE(dashboard_client->commandPowerOff());
+
+  // We will not get this answer
+  EXPECT_FALSE(dashboard_client->sendRequest("brake release", "non-existing-response"));
+
+  // A non-matching answer should throw an exception for this call
+  EXPECT_THROW(dashboard_client->sendRequestString("brake release", "non-existing-response"), UrException);
+
+  // Waiting for a non-matching answer should return false
+  // Internally we wait 100ms between each attempt, hence the 300ms wait time
+  EXPECT_FALSE(
+      dashboard_client->waitForReply("brake_release", "non-existing-response", std::chrono::milliseconds(300)));
+}
+
+TEST_F(DashboardClientTest, connecting_twoice_returns_false)
+{
+  std::unique_ptr<DashboardClient> dashboard_client;
```

Comment:
Why do you make a new dashboard client object and reset the communication again when that job already has been handle in the test class and its SetUP() function?